lib/RecommendationProvider.js

```diff
@@ -0,0 +1,126 @@
+/* globals exports, require, Task */
+"use strict";
+const {Cu} = require("chrome");
+const {setTimeout, clearTimeout} = require("sdk/timers");
+const simplePrefs = require("sdk/simple-prefs");
+
+const POCKET_API_URL = "pocket.endpoint";
+
+const DEFAULT_TIMEOUTS = {
+  pocketTimeout: 60000, // every 10 minutes, refresh the Pocket recommendations
+};
+
+Cu.import("resource://gre/modules/Task.jsm");
+Cu.importGlobalProperties(["fetch"]);
+
+function RecommendationProvider(previewProvider, options = {}) {
+  this.options = Object.assign({}, DEFAULT_TIMEOUTS, options);
+  this._recommendedContent = [];
+  this._blockedRecommendedContent = new Set();
+  this._currentRecommendation = null;
+  this._pocketEndpoint = simplePrefs.prefs[POCKET_API_URL];
+  this._previewProvider = previewProvider;
+  this._asyncUpdateRecommendations(this.options.pocketTimeout);
+}
+
+RecommendationProvider.prototype = {
+  _pocketTimeoutID: null,
+
+  /**
+    * If we need a recommendation, go get one
+    */
+  getRecommendation(refresh = false) {
+    // if we need a new recommendation purge the current recommendation and get a new one
+    if (refresh) {
+      this._currentRecommendation = null;
+    }
+    // if we are already displaying a recommendation, just show that one and don't get a new one
+    if (this._currentRecommendation) {
+      return this._currentRecommendation;
+    }
+    return this._getRandomRecommendation();
```

Comment:
Can we do

this._currentRecommendation = this._getRandomRecommendation();
return this._currentRecommendation;

here, and then _getRandomRecommendation has no side effects and just always return a random selection from available recommendations or null, and then the case of setting it to null is just automatic?